### DISCUSSION TITLE
[addon-a11y] is MIT licensing allowed?

### DISCUSSION THREAD

**Discussion by flo-sch** [storybookjs/storybook#15777] [Help]
Hello,

I have a question about the licensing of addon-a11y - not that I like the subject - since I received an internal question and am not a legal expert, just wondering if anyone has already checked that use-case?

1. addon-a11y has a [dependency to axe-core](https://github.com/storybookjs/storybook/blob/v6.3.6/addons/a11y/package.json#L56) which is strictly licensed under [MPL-2.0](https://github.com/dequelabs/axe-core/issues/2852)
2. addon-a11y itself seems to be [licensed under MIT](https://github.com/storybookjs/storybook/blob/v6.3.6/addons/a11y/package.json#L27)

According to my (non-expert) understanding of the MPL-2.0 terms about distribution, this seems not to be allowed.

> All distribution of Covered Software in Source Code Form, including any Modifications that You create or to which You contribute, must be under the terms of this License.

Unless perhaps it enters one of the exceptions to those terms? 🤷

Is addon-a11y safe to use under the MIT terms then (or is there an internal issue), or does the MPL-2.0 license apply to anyone including this addon in a Storybook project?

MPL-2.0 terms: https://www.mozilla.org/en-US/MPL/2.0/

**Comment by wtfiwtz** [comment]
My limited understanding is that if `axe` has been run within tests and your test code is not distributed/supplied (to another party, such as a customer or end-user) then it should be ok. However, you should be careful to observe the conditions of the MPL license, including if you need to publish a third-party license on your website, or leave the third-party notice in your code, particularly if it is derivative (bundled, minified JS) or otherwise downloadable.

Using `axe` in tests: https://github.com/storybookjs/storybook/blob/ce0b0205d73a58693617762594d1bdc0b7c3db73/code/addons/storyshots/storyshots-puppeteer/README.md#axetest
 
Under the MPL 2.0 terms...

```
3.3. Distribution of a Larger Work
You may create and distribute a Larger Work under terms of Your choice, provided that You also comply with the requirements of this License for the Covered Software. If the Larger Work is a combination of Covered Software with a work governed by one or more Secondary Licenses, and the Covered Software is not Incompatible With Secondary Licenses, this License permits You to additionally distribute such Covered Software under the terms of such Secondary License(s), so that the recipient of the Larger Work may, at their option, further distribute the Covered Software under the terms of either this License or such Secondary License(s).
```

and...

```
10.4. Distributing Source Code Form that is Incompatible With Secondary Licenses
If You choose to distribute Source Code Form that is Incompatible With Secondary Licenses under the terms of this version of the License, the notice described in Exhibit B of this License must be attached.

Exhibit B - “Incompatible With Secondary Licenses” Notice
This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
```

... seem important. So you/we should check for whether Axe supports Secondary Licenses.

Here's an interesting conversation in the Rust community... https://users.rust-lang.org/t/licensing-mit-apache-2-vs-mpl-2-0/46250/9

😵
